diff --git a/meson.build b/meson.build
index 69c1bb8..629aa73 100644
--- a/meson.build
+++ b/meson.build
@@ -297,14 +297,22 @@ if with_egl
    if cc.has_header('epoxy/egl.h', dependencies: epoxy_dep) and epoxy_dep.get_variable(pkgconfig: 'epoxy_has_egl') == '1'
       if with_host_windows or with_host_darwin
          have_egl = true
+      elif host_machine.system() == 'android'
+         have_egl = cc.has_header('sys/mman.h', required: require_egl)
+         if libdrm_dep.found()
+            gbm_dep = dependency('gbm', version: '>= 18.0.0', required: false)
+         endif
+         if gbm_dep.found() and have_egl
+            conf_data.set('ENABLE_GBM', 1)
+            conf_data.set('ENABLE_GBM_ALLOCATION', 1)
+         endif
       else
          if libdrm_dep.found()
             gbm_dep = dependency('gbm', version: '>= 18.0.0', required: require_egl)
          endif
          have_egl = gbm_dep.found() and cc.has_header('sys/mman.h', required: require_egl)
-         conf_data.set('ENABLE_GBM', have_egl)
-
          if have_egl
+            conf_data.set('ENABLE_GBM', 1)
             conf_data.set('ENABLE_GBM_ALLOCATION', 1)
          endif
       endif
diff --git a/src/vrend/vrend_winsys.c b/src/vrend/vrend_winsys.c
index d4c9b9e..63b3ef7 100644
--- a/src/vrend/vrend_winsys.c
+++ b/src/vrend/vrend_winsys.c
@@ -98,15 +98,17 @@ int vrend_winsys_init(uint32_t flags, int preferred_fd)
 
 void vrend_winsys_cleanup(void)
 {
-#ifdef ENABLE_GBM
+#ifdef HAVE_EPOXY_EGL_H
    if (use_context == CONTEXT_EGL) {
       virgl_egl_destroy(egl);
       egl = NULL;
       use_context = CONTEXT_NONE;
+#ifdef ENABLE_GBM
       if (gbm) {
          virgl_gbm_fini(gbm);
          gbm = NULL;
       }
+#endif
    } else if (use_context == CONTEXT_EGL_EXTERNAL) {
       free(egl);
       egl = NULL;
